### PR TITLE
pre-commit config

### PR DESCRIPTION
**Pull request by zmoon**
Based on xgcm's config but adding pyupgrade and facilitating the current longer line-length setting.

### CODE REVIEW

**Comment by zmoon** [comment]
@jbusecke I wanted to see what you thought of this config before I apply it. Also wait for #73 to be merged I suppose.

**Comment by jbusecke** [comment]
Hey @zmoon for this to work in the 'old way' you would have had to add a github workflow file, but I have started switching repos over to the [pre-commit service](https://pre-commit.ci). This is convenient since it does make linting corrections automatically!
Should we add this as part of this PR? If you agree, I would suggest to keep the line length as is (the bot should fix the code!), and I can activate the service.

**Comment by jbusecke** [comment]
I also think we can put this PR in front of #73, which might need some more iterations.

**Comment by zmoon** [comment]
> the [pre-commit service](https://pre-commit.ci)

Great, pre-commit ci is very nice. We should still do a `pre-commit run --all-files` here or later though.

> tried to make a PR on your PR

You should be able to checkout my PR and commit. I can do it though.

**Comment by zmoon** [comment]
I'm going to start resolving the failed checks here.

File: xmovie/core.py [hunk 1]
```diff
@@ -418,7 +399,7 @@ def save_frames_parallel(self, odir, parallel_compute_kwargs=dict()):
         elif type(da) is xr.Dataset:
             framedim_chunks = da.chunks[framedim]
         else:
-            raise(TypeError("`da` must be either an xarray.DataArray or xarray.Dataset"))
+            raise (TypeError("`da` must be either an xarray.DataArray or xarray.Dataset"))
```
Comment (zmoon) [inline xmovie/core.py:402]:
```suggestion
            raise TypeError("`da` must be either an xarray.DataArray or xarray.Dataset")
```

**Pull request merged**